ubi/src/os.rs

```diff
@@ -52,3 +52,29 @@ pub(crate) fn all_oses_re() -> Regex {
     )
     .unwrap()
 }
+
+fn all_arches_re() -> Regex {
+    Regex::new(
+        &[
+            regex!(r"(?i:(?:\b|_)(?:x86|x86_64|amd64)(?:\b|_))"),
+            regex!(r"(?i:(?:\b|_)(?:arm|arm64|aarch64)(?:\b|_))"),
+            regex!(r"(?i:(?:\b|_)(?:mips|mips64)(?:\b|_))"),
+            regex!(r"(?i:(?:\b|_)(?:ppc|ppc64)(?:\b|_))"),
+            regex!(r"(?i:(?:\b|_)(?:s390x)(?:\b|_))"),
+        ]
+        .iter()
+        .map(|r| format!("({})", r.as_str()))
+        .join("|"),
+    )
+    .unwrap()
+}
+
+pub(crate) static PLATFORM_RE: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+        &[all_oses_re(), all_arches_re()]
+            .iter()
+            .map(|r| format!("({})", r.as_str()))
+            .join("|"),
+    )
+    .unwrap()
+});
```

Comment:
I think this can be constructed in the `extension` module by importing the relevant fns from `os` and `arch`.

Reply:
idk how significant it is but won't that involve recompiling the final regex many times?

Reply:
I thought the use of `Lazy` would mean it's only compiled once, right?

Reply:
Or rather, can't `Lazy` also be used in the `extension` module? Maybe there's some issue with doing that?

Reply:
all_oses_re is not wrapped in Lazy and gets called many times

Reply:
Ah, right. I just pushed a change to make both of these `static Lazy<Regex>` vars instead of functions.